src/main/java/org/opensearch/ad/ml/CheckpointDao.java

```diff
@@ -428,17 +428,30 @@ private void logFailure(BulkByScrollResponse response, String detectorID) {
     /**
      * Read a checkpoint from the index and return the EntityModel object
      * @param modelId Model Id
-     * @param listener Listener to return the EntityModel object
+     * @param listener Listener to return a pair of entity model and its last checkpoint time
      */
     public void restoreModelCheckpoint(String modelId, ActionListener<Optional<Entry<EntityModel, Instant>>> listener) {
-        clientUtil.<GetRequest, GetResponse>asyncRequest(new GetRequest(indexName, modelId), client::get, ActionListener.wrap(response -> {
-            Optional<Map<String, Object>> checkpointString = processRawCheckpoint(response);
-            if (checkpointString.isPresent()) {
-                listener.onResponse(fromEntityModelCheckpoint(checkpointString.get(), modelId));
-            } else {
-                listener.onResponse(Optional.empty());
-            }
-        }, listener::onFailure));
+        clientUtil
+            .<GetRequest, GetResponse>asyncRequest(
+                new GetRequest(indexName, modelId),
+                client::get,
+                ActionListener.wrap(response -> { listener.onResponse(processGetResponse(response, modelId)); }, listener::onFailure)
+            );
+    }
+
+    /**
+     * Process a checkpoint GetResponse and return the EntityMOdel object
+     * @param response Checkpoint Index GetResponse
+     * @param modelId  Model Id
+     * @return a pair of entity model and its last checkpoint time
+     */
+    public Optional<Entry<EntityModel, Instant>> processGetResponse(GetResponse response, String modelId) {
```

Comment:
Does this need to be public?

Reply:
It's used in `CheckpointReadWorker` in #2